students/VIK/04_METAPROGRAMMING/mailroom.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env python3
+
+"""********************************************************************************************************************
+         TITLE: UW PYTHON 220 - Lesson 04 - Activity
+     SUB TITLE: Mailroom - Metaprogramming
+       CREATOR: PydPiper
+  DATE CREATED: 4/28/18
+ LAST MODIFIED: 5/5/18
+   DESCRIPTION: Convert the 210 mailroom programming's database into JSON save/load format via metaprogramming
+********************************************************************************************************************"""
+
+""" IMPORTS """
+import json_save_meta as js
+
+
+class Data(js.JsonSaveable):
+    # data_dict = {}
+    # you have to declare a class attribute before initializing js.Dict()
+    data_dict = js.Dict().default
+    data = js.Dict()
+
+    def __init__(self, name=""):
+        # current name
+        self.name = name
+        # add to dictionary
+        if name != "":
+            try:
+                self.data_dict[name]
+            except KeyError:
+                self.data_dict.update({name: [0, 0]})
+
+    def donations(self, val=0):
+        # print("data donations")
+        # return current donation amount for name
+        c_don = self.data_dict[self.name][0]
+        c_gift = self.data_dict[self.name][1]
+        if val == 0: add_gift = 0
+        else: add_gift = 1
```

Comment:
try to break these out into 4 lines.  It just reads very weird in 2 lines.  i don't know, maybe its just me :)